module/VuFind/src/VuFind/RecordDriver/EDS.php

```diff
@@ -431,19 +443,19 @@ public function getLinkedFullTextLink()
     }
 
     /**
-     * Get the subject data of the record.
+     * Get the subject headings as a flat array of strings.
      *
-     * @return string
+     * @return array Subject headings
      */
-    public function getItemsSubjects()
+    public function getAllSubjectHeadingsFlattened()
     {
-        $subjects = array_map(
+        $subject_arrays = array_map(
             function ($data) {
-                return $data['Data'];
+                return explode(', ', rtrim(strip_tags($data['Data']), '.'));
```

Comment:
Is it safe to explode on comma, or would we be better off doing some kind of regular expression extraction? I see from the example that there's a comma-separated list of `<a>` tags, but what happens if there's a comma inside the tag? It seems to me that this might be possible, though I can't be totally sure....

Reply:
A valid concern. The problem is that the data we get from EDS is a mess. For instance, major subjects are separated with br tags, while minor subjects are separated with ";". VuFind is currently not parsing that correctly, so major and minor subjects are not displayed in the same way on the record page. Using the tags to try to separate entries is a good idea, maybe we can split on "`>; <`" or "`>, <`" instead. I wish we had better parsing of the data, structuring it and removing all HTML, but I don't want to get into that in the scope of that PR.

Reply:
What if you did something like `preg_replace('/>\s*[;,]\s*', '>|', $str)` and then `explode('|', strip_tags($str))` on the result? That might cover more cases without breaking the integrity of the HTML in a way that would mess up the tag stripping...

Reply:
I changed the implementation using your idea.